lib/Sema/CSDiag.cpp

```diff
@@ -3443,15 +3443,37 @@ Expr *FailureDiagnosis::typeCheckChildIndependently(
       if (FT->isAutoClosure())
         convertType = FT->getResult();
 
-    // Replace archetypes and type parameters with UnresolvedType.
-    convertType = replaceTypeParametersWithUnresolved(convertType);
-    convertType = replaceTypeVariablesWithUnresolved(convertType);
-    
+    bool shouldNullify = false;
```

Comment:
And just to clarify a bit, `objectType->hasUnresolvedType()` is not appropriate in this case standalone, because if it's in a function for example or inout type we still want to preserve it's _skeleton_ because that helps to diagnose _inout_ argument issues. Complete nullification is only appropriate for generic types with unresolved types or standalone archetypes because that's going to give sub-expression solver a chance to try and compute type as it sees and higher level code is going to check it, which avoids error messages like `cannot convert (_) -> _ to (Int) -> Void` or similar.